--- origsrc/hexchat-2.10.0/plugins/perl/perl.c	2014-06-01 11:11:26.000000000 -0500
+++ src/hexchat-2.10.0/plugins/perl/perl.c	2014-06-06 14:15:53.441515300 -0500
@@ -623,7 +623,6 @@ print_cb (char *word[], void *userdata)
  *
  */
 
-static
 XS (XS_HexChat_register)
 {
 	char *name, *version, *desc, *filename;
@@ -648,7 +647,6 @@ XS (XS_HexChat_register)
 
 
 /* HexChat::print(output) */
-static
 XS (XS_HexChat_print)
 {
 
@@ -664,7 +662,6 @@ XS (XS_HexChat_print)
 	XSRETURN_EMPTY;
 }
 
-static
 XS (XS_HexChat_emit_print)
 {
 	char *event_name;
@@ -718,7 +715,6 @@ XS (XS_HexChat_emit_print)
 	}
 }
 
-static
 XS (XS_HexChat_send_modes)
 {
 	AV *p_targets = NULL;
@@ -771,7 +767,6 @@ XS (XS_HexChat_send_modes)
 		free (targets);
 	}
 }
-static
 XS (XS_HexChat_get_info)
 {
 	SV *temp = NULL;
@@ -810,7 +805,6 @@ XS (XS_HexChat_get_info)
 	}
 }
 
-static
 XS (XS_HexChat_context_info)
 {
 	const char *const *fields;
@@ -824,7 +818,6 @@ XS (XS_HexChat_context_info)
 	XSRETURN (1);
 }
 
-static
 XS (XS_HexChat_get_prefs)
 {
 	const char *str;
@@ -862,7 +855,6 @@ XS (XS_HexChat_get_prefs)
 }
 
 /* HexChat::Internal::hook_server(name, priority, callback, userdata) */
-static
 XS (XS_HexChat_hook_server)
 {
 
@@ -903,7 +895,6 @@ XS (XS_HexChat_hook_server)
 }
 
 /* HexChat::Internal::hook_command(name, priority, callback, help_text, userdata) */
-static
 XS (XS_HexChat_hook_command)
 {
 	char *name;
@@ -952,7 +943,6 @@ XS (XS_HexChat_hook_command)
 }
 
 /* HexChat::Internal::hook_print(name, priority, callback, [userdata]) */
-static
 XS (XS_HexChat_hook_print)
 {
 
@@ -991,7 +981,6 @@ XS (XS_HexChat_hook_print)
 }
 
 /* HexChat::Internal::hook_timer(timeout, callback, userdata) */
-static
 XS (XS_HexChat_hook_timer)
 {
 	int timeout;
@@ -1030,7 +1019,6 @@ XS (XS_HexChat_hook_timer)
 }
 
 /* HexChat::Internal::hook_fd(fd, callback, flags, userdata) */
-static
 XS (XS_HexChat_hook_fd)
 {
 	int fd;
@@ -1083,7 +1071,6 @@ XS (XS_HexChat_hook_fd)
 	}
 }
 
-static
 XS (XS_HexChat_unhook)
 {
 	hexchat_hook *hook;
@@ -1119,7 +1106,6 @@ XS (XS_HexChat_unhook)
 }
 
 /* HexChat::Internal::command(command) */
-static
 XS (XS_HexChat_command)
 {
 	char *cmd = NULL;
@@ -1135,7 +1121,6 @@ XS (XS_HexChat_command)
 	XSRETURN_EMPTY;
 }
 
-static
 XS (XS_HexChat_find_context)
 {
 	char *server = NULL;
@@ -1191,7 +1176,6 @@ XS (XS_HexChat_find_context)
 	}
 }
 
-static
 XS (XS_HexChat_get_context)
 {
 	dXSARGS;
@@ -1202,7 +1186,6 @@ XS (XS_HexChat_get_context)
 	}
 }
 
-static
 XS (XS_HexChat_set_context)
 {
 	hexchat_context *ctx;
@@ -1215,7 +1198,6 @@ XS (XS_HexChat_set_context)
 	}
 }
 
-static
 XS (XS_HexChat_nickcmp)
 {
 	dXSARGS;
@@ -1227,7 +1209,6 @@ XS (XS_HexChat_nickcmp)
 	}
 }
 
-static
 XS (XS_HexChat_get_list)
 {
 	SV *name;
@@ -1268,7 +1249,6 @@ XS (XS_HexChat_get_list)
 	}
 }
 
-static
 XS (XS_HexChat_Embed_plugingui_remove)
 {
 	void *gui_entry;
@@ -1282,7 +1262,6 @@ XS (XS_HexChat_Embed_plugingui_remove)
 	XSRETURN_EMPTY;
 }
 
-static
 XS (XS_HexChat_plugin_pref_set)
 {
 	dMARK;
@@ -1292,7 +1271,6 @@ XS (XS_HexChat_plugin_pref_set)
 													SvPV_nolen (ST (1))));
 }
 
-static
 XS (XS_HexChat_plugin_pref_get)
 {
 	int result;
@@ -1309,7 +1287,6 @@ XS (XS_HexChat_plugin_pref_get)
 	XSRETURN_UNDEF;
 }
 
-static
 XS (XS_HexChat_plugin_pref_delete)
 {
 	dMARK;
@@ -1318,7 +1295,6 @@ XS (XS_HexChat_plugin_pref_delete)
 	XSRETURN_IV ((IV) hexchat_pluginpref_delete (ph, SvPV_nolen (ST (0))));
 }
 
-static
 XS (XS_HexChat_plugin_pref_list)
 {
 	char list[4096];
