--- origsrc/hexchat-2.10.0/plugins/perl/Makefile.am	2014-06-06 14:19:34.874180500 -0500
+++ src/hexchat-2.10.0/plugins/perl/Makefile.am	2014-06-06 14:22:21.250696700 -0500
@@ -15,4 +15,4 @@ CLEANFILES = hexchat.pm.h irc.pm.h
 hexchat.pm.h irc.pm.h: lib/HexChat.pm lib/Xchat.pm lib/HexChat/Embed.pm \
 	lib/HexChat/List/Network.pm lib/HexChat/List/Network/Entry.pm \
 	lib/HexChat/List/Network/AutoJoin.pm lib/IRC.pm
-	perl generate_header
+	cd $(srcdir); perl generate_header
--- origsrc/hexchat-2.10.0/src/common/Makefile.am	2014-06-01 11:11:26.000000000 -0500
+++ src/hexchat-2.10.0/src/common/Makefile.am	2014-06-06 14:23:07.910365500 -0500
@@ -69,8 +69,8 @@ libhexchatcommon_a_CFLAGS = $(LIBPROXY_C
 
 textenums.h: textevents.h
 
-textevents.h: textevents.in make-te
-	$(AM_V_GEN) ./make-te < textevents.in > textevents.h 2> textenums.h
+textevents.h: textevents.in make-te$(EXEEXT)
+	$(AM_V_GEN) ./make-te < $(srcdir)/textevents.in > textevents.h 2> textenums.h
 
 marshal.h: marshalers.list
 	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=_hexchat_marshal --header $(srcdir)/marshalers.list > $@
--- origsrc/hexchat-2.10.0/src/common/hexchat.h	2014-06-01 11:11:26.000000000 -0500
+++ src/hexchat-2.10.0/src/common/hexchat.h	2014-06-06 14:22:21.253696800 -0500
@@ -20,7 +20,7 @@
 #ifdef WIN32
 #include "../../config-win32.h"
 #else
-#include "../../config.h"
+#include "config.h"
 #endif
 
 #include <glib.h>
--- origsrc/hexchat-2.10.0/src/fe-gtk/Makefile.am	2014-06-01 11:11:26.000000000 -0500
+++ src/hexchat-2.10.0/src/fe-gtk/Makefile.am	2014-06-06 14:22:21.255697000 -0500
@@ -2,7 +2,7 @@ localedir = $(datadir)/locale
 
 bin_PROGRAMS = hexchat
 
-AM_CPPFLAGS = $(GUI_CFLAGS) -DLOCALEDIR=\"$(localedir)\"
+AM_CPPFLAGS = $(GUI_CFLAGS) -I../common -DLOCALEDIR=\"$(localedir)\"
 
 hexchat_LDADD = ../common/libhexchatcommon.a $(GUI_LIBS)
 
@@ -32,5 +32,5 @@ hexchat_SOURCES = ascii.c banlist.c chan
 	rawlog.c resources.c servlistgui.c setup.c $(iso_codes_c) \
 	sexy-spell-entry.c textgui.c urlgrab.c userlistgui.c xtext.c
 
-resources.c: ../../data/hexchat.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=../../data --generate-dependencies ../../data/hexchat.gresource.xml)
-	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=../../data --generate-source $<
+resources.c: $(top_srcdir)/data/hexchat.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(top_srcdir)/data --generate-dependencies $(top_srcdir)/data/hexchat.gresource.xml)
+	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(top_srcdir)/data --generate-source $<
