--- origsrc/hexchat-2.12.1/configure.ac	2016-05-01 13:34:42.000000000 -0500
+++ src/hexchat-2.12.1/configure.ac	2016-05-02 11:23:16.722715100 -0500
@@ -644,6 +644,7 @@ AX_APPEND_COMPILE_FLAGS([\
 ])
 
 AX_APPEND_LINK_FLAGS([ \
+	-fstack-protector-strong \
 	-pie \
 	-Wl,-z,relro \
 	-Wl,-z,now \
--- origsrc/hexchat-2.12.1/src/common/hexchat-plugin.h	2015-04-29 13:18:57.000000000 -0500
+++ src/hexchat-2.12.1/src/common/hexchat-plugin.h	2016-05-02 11:10:41.780563800 -0500
@@ -406,7 +406,7 @@ int
 hexchat_pluginpref_list (hexchat_plugin *ph,
 		char *dest);
 
-#if !defined(PLUGIN_C) && defined(WIN32)
+#if !defined(PLUGIN_C) && (defined(WIN32) || defined(__CYGWIN__))
 #ifndef HEXCHAT_PLUGIN_HANDLE
 #define HEXCHAT_PLUGIN_HANDLE (ph)
 #endif
