--- origsrc/hexchat-2.12.3/src/fe-gtk/sexy-spell-entry.c	2016-04-12 17:10:32.000000000 -0500
+++ src/hexchat-2.12.3/src/fe-gtk/sexy-spell-entry.c	2017-01-18 12:54:52.786163400 -0600
@@ -161,7 +161,11 @@ initialize_enchant (void)
 	gpointer funcptr;
 
 
+#ifdef __CYGWIN__
+	enchant = g_module_open("cygenchant-1.dll", 0);
+#else
 	enchant = g_module_open("libenchant."G_MODULE_SUFFIX, 0);
+#endif
 	if (enchant == NULL)
 	{
 #ifndef WIN32
