--- origsrc/hexchat-2.9.6.1/plugins/checksum/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/checksum/Makefile.am	2014-05-02 10:27:55.274064500 -0500
@@ -2,6 +2,6 @@ libdir = $(hexchatlibdir)
 
 lib_LTLIBRARIES = checksum.la
 checksum_la_SOURCES = checksum.c
-checksum_la_LDFLAGS = -avoid-version -module 
-checksum_la_LIBADD = 
+checksum_la_LDFLAGS = -avoid-version -module -no-undefined
+checksum_la_LIBADD = $(COMMON_LIBS)
 AM_CPPFLAGS = $(COMMON_CFLAGS) -I$(srcdir)/../../src/common
--- origsrc/hexchat-2.9.6.1/plugins/doat/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/doat/Makefile.am	2014-05-02 10:28:08.370727500 -0500
@@ -2,7 +2,7 @@ libdir = $(hexchatlibdir)
 
 lib_LTLIBRARIES = doat.la
 doat_la_SOURCES = doat.c
-doat_la_LDFLAGS = -avoid-version -module 
-doat_la_LIBADD = 
+doat_la_LDFLAGS = -avoid-version -module -no-undefined
+doat_la_LIBADD = $(COMMON_LIBS)
 AM_CPPFLAGS = $(COMMON_CFLAGS) -I$(srcdir)/../../src/common
 
--- origsrc/hexchat-2.9.6.1/plugins/fishlim/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/fishlim/Makefile.am	2014-05-02 10:28:14.389491800 -0500
@@ -4,6 +4,6 @@ libdir = $(hexchatlibdir)
 
 lib_LTLIBRARIES = fishlim.la
 fishlim_la_SOURCES = fish.c irc.c keystore.c misc.c plugin_hexchat.c
-fishlim_la_LDFLAGS = -avoid-version -module 
-fishlim_la_LIBADD = 
+fishlim_la_LDFLAGS = -avoid-version -module -no-undefined
+fishlim_la_LIBADD = $(COMMON_LIBS)
 AM_CPPFLAGS = $(COMMON_CFLAGS) -I$(srcdir)/../../src/common
--- origsrc/hexchat-2.9.6.1/plugins/mailcheck/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/mailcheck/Makefile.am	2014-05-02 10:28:32.004228600 -0500
@@ -19,7 +19,8 @@ libdir = $(hexchatdir)/plugins
 
 lib_LTLIBRARIES = mailcheck.la
 mailcheck_la_SOURCES = mailcheck.c
-mailcheck_la_LDFLAGS = -avoid-version -module 
+mailcheck_la_LDFLAGS = -avoid-version -module -no-undefined
+mailcheck_la_LIBADD = $(COMMON_LIBS)
 INCLUDES = $(COMMON_CFLAGS) $(PLUGIN_INCLUDES)
 
 DISTCLEANFILES = pg_dir/*
--- origsrc/hexchat-2.9.6.1/plugins/perl/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/perl/Makefile.am	2014-05-02 10:30:11.887912200 -0500
@@ -6,8 +6,8 @@ libdir = $(hexchatlibdir)
 
 lib_LTLIBRARIES = perl.la
 perl_la_SOURCES = perl.c
-perl_la_LDFLAGS = -avoid-version -module 
-perl_la_LIBADD = $(PERL_LDFLAGS)
+perl_la_LDFLAGS = -avoid-version -module -no-undefined
+perl_la_LIBADD = $(PERL_LDFLAGS) $(COMMON_LIBS)
 BUILT_SOURCES = xchat.pm.h irc.pm.h
 #CFLAGS = @CFLAGS@ -Wno-unused
 AM_CPPFLAGS = $(PERL_CFLAGS) $(COMMON_CFLAGS) -I$(srcdir)/../../src/common
--- origsrc/hexchat-2.9.6.1/plugins/python/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/python/Makefile.am	2014-05-02 10:27:24.949213700 -0500
@@ -4,7 +4,7 @@ libdir = $(hexchatlibdir)
 
 lib_LTLIBRARIES = python.la
 python_la_SOURCES = python.c
-python_la_LDFLAGS = -avoid-version -module 
-python_la_LIBADD = $(PY_LIBS)
+python_la_LDFLAGS = -avoid-version -module -no-undefined
+python_la_LIBADD = $(PY_LIBS) $(COMMON_LIBS)
 AM_CPPFLAGS = $(PY_CFLAGS) $(COMMON_CFLAGS) -I$(srcdir)/../../src/common
 
--- origsrc/hexchat-2.9.6.1/plugins/xdcc/Makefile.am	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/plugins/xdcc/Makefile.am	2014-05-02 10:28:55.327190300 -0500
@@ -2,6 +2,7 @@ libdir = $(hexchatdir)
 
 lib_LTLIBRARIES = xdcc.la
 xdcc_la_SOURCES = xdcc.c
-xdcc_la_LDFLAGS = -avoid-version -module 
+xdcc_la_LDFLAGS = -avoid-version -module -no-undefined
+xdcc_la_LIBADD = $(COMMON_LIBS)
 INCLUDES = $(COMMON_CFLAGS) -I$(srcdir)/..
 
--- origsrc/hexchat-2.9.6.1/src/common/hexchat-plugin.h	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/src/common/hexchat-plugin.h	2014-05-02 10:29:40.485424600 -0500
@@ -390,7 +390,7 @@ int
 hexchat_pluginpref_list (hexchat_plugin *ph,
 		char *dest);
 
-#if !defined(PLUGIN_C) && defined(WIN32)
+#if !defined(PLUGIN_C) && (defined(WIN32) || defined(__CYGWIN__))
 #ifndef HEXCHAT_PLUGIN_HANDLE
 #define HEXCHAT_PLUGIN_HANDLE (ph)
 #endif
