--- origsrc/hexchat-2.9.6.1/src/common/util.c	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/src/common/util.c	2014-05-02 10:19:26.809997700 -0500
@@ -28,9 +28,11 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
+#if defined (WIN32) || defined (__CYGWIN__)
+#include <process.h>
+#endif
 #ifdef WIN32
 #include <sys/timeb.h>
-#include <process.h>
 #include <io.h>
 #include <pango/pangocairo.h>		/* for find_font() */
 #include "../dirent/dirent-win32.h"
@@ -1393,7 +1395,7 @@ util_exec (const char *cmd)
 {
 	char **argv;
 	int argc;
-#ifndef WIN32
+#if !defined (WIN32) && !defined(__CYGWIN__)
 	int pid;
 	int fd;
 #endif
@@ -1401,7 +1403,7 @@ util_exec (const char *cmd)
 	if (my_poptParseArgvString (cmd, &argc, &argv) != 0)
 		return -1;
 
-#ifndef WIN32
+#if !defined (WIN32) && !defined(__CYGWIN__)
 	pid = fork ();
 	if (pid == -1)
 		return -1;
@@ -1426,7 +1428,7 @@ util_exec (const char *cmd)
 int
 util_execv (char * const argv[])
 {
-#ifndef WIN32
+#if !defined (WIN32) && !defined(__CYGWIN__)
 	int pid, fd;
 
 	pid = fork ();
