--- origsrc/hexchat-2.9.6.1/configure.ac	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/configure.ac	2014-05-02 10:21:48.069435400 -0500
@@ -32,6 +32,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GET
 
 ALL_LINGUAS="af am ast az be bg ca cs da de el en_GB es et eu fi fr gl gu hi hu id it ja kn ko lt lv mk ml ms nb nl no pa pl pt pt_BR ru rw sk sl sq sr sv th uk vi wa zh_CN zh_TW"
 AM_GNU_GETTEXT
+AM_GNU_GETTEXT_VERSION(0.18.2)
 
 dnl displaced from acconfig.h
 AH_VERBATIM([OLD_PERL],[#undef OLD_PERL])
