--- origsrc/hexchat-2.9.6.1/src/common/outbound.c	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/src/common/outbound.c	2014-05-02 10:23:53.679385800 -0500
@@ -2525,7 +2525,7 @@ cmd_load (struct session *sess, char *tb
 
 #ifdef USE_PLUGIN
 	len = strlen (word[2]);
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 	if (len > 4 && g_ascii_strcasecmp (".dll", word[2] + len - 4) == 0)
 #else
 #if defined(__hpux)
@@ -3471,7 +3471,7 @@ cmd_unload (struct session *sess, char *
 	int len, by_file = FALSE;
 
 	len = strlen (word[2]);
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 	if (len > 4 && g_ascii_strcasecmp (word[2] + len - 4, ".dll") == 0)
 #else
 #if defined(__hpux)
@@ -3505,7 +3505,7 @@ cmd_reload (struct session *sess, char *
 	int len, by_file = FALSE;
 
 	len = strlen (word[2]);
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 	if (len > 4 && g_ascii_strcasecmp (word[2] + len - 4, ".dll") == 0)
 #else
 #if defined(__hpux)
--- origsrc/hexchat-2.9.6.1/src/common/plugin.c	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/src/common/plugin.c	2014-05-02 10:24:50.372084900 -0500
@@ -499,15 +499,16 @@ plugin_auto_load (session *sess)
 	for_files (HEXCHATLIBDIR, "hcsysinfo.dll", plugin_auto_load_cb);
 
 	for_files (sub_dir, "*.dll", plugin_auto_load_cb);
-#else
-#if defined(__hpux)
+#elif defined(__CYGWIN__)
+	for_files (HEXCHATLIBDIR, "*.dll", plugin_auto_load_cb);
+	for_files (sub_dir, "*.dll", plugin_auto_load_cb);
+#elif defined(__hpux)
 	for_files (HEXCHATLIBDIR, "*.sl", plugin_auto_load_cb);
 	for_files (sub_dir, "*.sl", plugin_auto_load_cb);
 #else
 	for_files (HEXCHATLIBDIR, "*.so", plugin_auto_load_cb);
 	for_files (sub_dir, "*.so", plugin_auto_load_cb);
 #endif
-#endif
 
 	g_free (sub_dir);
 }
--- origsrc/hexchat-2.9.6.1/src/fe-gtk/plugingui.c	2013-09-11 17:18:19.000000000 -0500
+++ src/hexchat-2.9.6.1/src/fe-gtk/plugingui.c	2014-05-02 10:25:33.454055600 -0500
@@ -161,7 +161,7 @@ plugingui_load (void)
 	sub_dir = g_build_filename (get_xdir(), "addons", NULL);
 
 	gtkutil_file_req (_("Select a Plugin or Script to load"), plugingui_load_cb, current_sess,
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 							sub_dir, "*.dll;*.lua;*.pl;*.py;*.tcl;*.js", FRF_FILTERISINITIAL|FRF_EXTENSIONS);
 #else
 							sub_dir, "*.so;*.lua;*.pl;*.py;*.tcl;*.js", FRF_FILTERISINITIAL|FRF_EXTENSIONS);
@@ -190,7 +190,7 @@ plugingui_unload (GtkWidget * wid, gpoin
 		return;
 
 	len = strlen (file);
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 	if (len > 4 && g_ascii_strcasecmp (file + len - 4, ".dll") == 0)
 #else
 #if defined(__hpux)
